pandas/tests/test_common.py

```diff
@@ -245,3 +245,19 @@ def test_str_size():
     ]
     result = subprocess.check_output(call).decode()[-4:-1].strip("\n")
     assert int(result) == int(expected)
+
+
+def test_bz2_missing_import():
+    # Check whether bz2 missing import is handled correctly (issue #53857)
+    code = (
+        "import pytest\n"
+        "sys.modules.pop('bz2', None)\n"
+        "import pandas\n"
+        "assert 'bz2' not in sys.modules\n"
+        "from pandas.compat import get_bz2_file\n"
+        "msg = 'bz2 module not available.'\n"
+        "with pytest.raises(RuntimeError, match=msg):\n"
+        "\tget_bz2_file()"
```

Comment:
This will fail on our CI because bz2 is installed, yes? I think L255-256 is sufficient here.

Reply:
Hm, I had thought the `sys.modules.pop("bz2", None)` would have "hidden" the `bz2` module and stopped the import somehow, thus making `get_bz2_file()` throw an error. What is the purpose of this unittest? @lithomas1

Reply:
`sys.modules` is a cache; if you remove module XYZ from it and then do `import XYZ`, Python will try to import XYZ.

`has_bz2` is determined by trying to import `bz2`; I believe there is no way to get that to fail if it exists. I think the purpose of this test is to ensure that `import pandas` alone does not _try to_ import `bz2`; rather it is only imported if it is needed.

Reply:
Ah, I see, thanks. In that case we should also set `pandas.compat.has_bz2 = False`; then we can check that particular error.

Reply:
I think you can just use pytest.monkeypatch here and call delattr on the `bz2` module

https://docs.pytest.org/en/6.2.x/monkeypatch.html

Reply:
I think you're trying to do two things with this test.

The first thing with ``sys.modules`` is to check that ``bz2`` is not imported (which is what I mentioned).

The second thing is testing the exception.
I think what you can do is set ``sys.modules['bz2']`` to ``None``, to get an ``ImportError``.
For me, I get
```
>>> import sys
>>> sys.modules['bz2'] = None
>>> import pandas as pd
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/thomasli/opt/miniconda3/lib/python3.9/site-packages/pandas/__init__.py", line 22, in <module>
    from pandas.compat import is_numpy_dev as _is_numpy_dev  # pyright: ignore # noqa:F401
  File "/Users/thomasli/opt/miniconda3/lib/python3.9/site-packages/pandas/compat/__init__.py", line 24, in <module>
    import pandas.compat.compressors
  File "/Users/thomasli/opt/miniconda3/lib/python3.9/site-packages/pandas/compat/compressors.py", line 7, in <module>
    import bz2
ModuleNotFoundError: import of bz2 halted; None in sys.modules
>>> import awef
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'awef
```

Because both of these things are mucking around with sys.modules, I think you should split them into separate tests.

Reply:
Updated; let me know if this works.

Reply:
Passing locally. I just had to figure out how to use an indented block when passing to python via the `-c` flag.